tests/MongoDB.Bson.Tests/IO/JsonReaderTests.cs

```diff
@@ -919,6 +1116,12 @@ public void TestObjectIdStrict()
         [Theory]
         [InlineData("{ $regex : \"abc\", $options : \"i\" }", "abc", "i")]
         [InlineData("{ $regex : \"abc/\", $options : \"i\" }", "abc/", "i")]
+        [InlineData("{ $regularExpression : { pattern : \"abc\", options : \"i\" } }", "abc", "i")]
+        [InlineData("{ $regularExpression : { pattern : \"abc/\", options : \"i\" } }", "abc/", "i")]
+        [InlineData("{ $regularExpression : { options : \"i\", pattern : \"abc\" } }", "abc", "i")]
+        [InlineData("{ $regularExpression : { options : \"i\", pattern : \"abc/\" } }", "abc/", "i")]
```

Comment:
Test the new canonical version of $regularExpression

Reply:
Test with empty string for options and pattern?  (I see similar tests in the BSON corpus).

        {
            "description": "empty regex with no options",
            "canonical_bson": "0A0000000B6100000000",
            "canonical_extjson": "{\"a\" : {\"$regularExpression\" : { \"pattern\": \"\", \"options\" : \"\"}}}"
        },
        {
            "description": "regex without options",
            "canonical_bson": "0D0000000B6100616263000000",
            "canonical_extjson": "{\"a\" : {\"$regularExpression\" : { \"pattern\": \"abc\", \"options\" : \"\"}}}"
        },